clang/lib/Index/IndexUnitReader.cpp

```diff
@@ -374,6 +381,13 @@ void IndexUnitReaderImpl::constructFilePath(SmallVectorImpl<char> &PathBuf,
   sys::path::append(PathBuf,
                     getPathFromBuffer(Path.Dir.Offset, Path.Dir.Size),
                     getPathFromBuffer(Path.Filename.Offset, Path.Filename.Size));
+  if (Path.PrefixKind == UNIT_PATH_PREFIX_NONE && !Remapper.empty()) {
+    SmallString<256> PathStr;
+    PathStr.assign(PathBuf);
+    std::string Remapped = Remapper.remapPath(PathStr.str());
+    PathBuf.clear();
+    PathBuf.append(Remapped.begin(), Remapped.end());
```

Comment:
Can't we just pass in `PathBuf` to `remapPath`? Is also a little unfortunate that we have to allocate a new string even if the path won't be remapped. Perhaps we could have a `remapPath` that takes a buffer rather than a `StringRef`? Then we could just pass `PathBuf` down.

We could keep the existing for other simple cases.